editor/window_wrapper.h

```diff
@@ -0,0 +1,99 @@
+/**************************************************************************/
+/*  window_wrapper.h                                                      */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef WINDOW_WRAPPER_H
+#define WINDOW_WRAPPER_H
+
+#include "scene/gui/margin_container.h"
+#include "scene/gui/menu_button.h"
+
+class Window;
+
+class WindowWrapper : public MarginContainer {
+	GDCLASS(WindowWrapper, MarginContainer);
+
+	Control *wrapped_control = nullptr;
+	MarginContainer *margins = nullptr;
```

Comment:
WindowWrapper is already a MarginContainer, so what is this for?

Reply:
WindowWrapper is a MarginContainer just because I need a container and MarginContainer is the simplest.
The real window margins are applied using `margins`, which has `window` as parent.

Reply:
But you can just set wrapper's margins instead of adding another control.

Reply:
Let me explain the node layout of WindowWrapper.

```
WindowWrapper:
    wrapped_control (when docked)
    window
        window_background
        margins
            wrapped_control (when undocked)
```

As you can see, WindowWrapper and margins are in different windows, and used in different situations.